examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/PackageCardAdapter.kt

```diff
@@ -48,19 +76,56 @@ class PackageCardAdapter(
                 currentPackage.packageType.toString()
             }
 
-            binding.packageDetailsJsonObject.detail = ""
-                // TODOBC5 currentPackage.product.originalJson.toString(JSON_FORMATTER_INDENT_SPACES)
+            binding.packageDetailsJsonObject.detail = product.googleProduct?.productDetails?.toString() ?: "TODO Amazon"
+
+            bindPurchaseOptions(product)
 
             binding.root.setOnClickListener {
                 with(binding.packageDetailsContainer) {
                     visibility = if (visibility == View.GONE) View.VISIBLE else View.GONE
                 }
             }
         }
+
+        private fun bindPurchaseOptions(product: StoreProduct) {
+            binding.packagePurchaseOptionGroup.removeAllViews()
+            val numberOfPurchaseOptions = product.purchaseOptions.size
+            product.purchaseOptions.forEach { purchaseOption ->
+                val radioButton = RadioButton(binding.root.context).apply {
+                    text = purchaseOption.toButtonString()
+                    tag = purchaseOption
+                }
+                binding.packagePurchaseOptionGroup.addView(radioButton)
+                if (numberOfPurchaseOptions == 1) binding.packagePurchaseOptionGroup.check(radioButton.id)
+            }
+        }
+
+        private fun getSelectedPurchaseOption(): PurchaseOption? {
+            val selectedButtonId = binding.packagePurchaseOptionGroup.checkedRadioButtonId
+            return binding.packagePurchaseOptionGroup.children
+                .filter { it.id == selectedButtonId }
+                .firstOrNull()
+                ?.tag as? PurchaseOption
+        }
+
+        private fun validateStartPurchase(product: StoreProduct): String? {
+            if (product.type == ProductType.SUBS
+                && binding.packagePurchaseOptionGroup.checkedRadioButtonId == nothingCheckedIndex) {
+                return "Please choose purchase option first"
```

Comment:
idk if it might be easier to just default the first purchaseoption to be checked...but i suppose this more closely mimics our api ;)

Reply:
Probably yeah... But if there is a choice I think it's safer to make an intentional decision rather than just choose one at random. We could autoselect the base plan but well or if we end up adding purchase options to OTP as suggested in #685 this should be removed as well. Will merge this like this for now and we can iterate more in the future.